x-pack/plugins/observability_solution/synthetics/server/alert_rules/status_rule/monitor_status_rule.ts

```diff
@@ -104,59 +80,103 @@ export const registerSyntheticsStatusCheckRule = (
         savedObjectsClient,
         scopedClusterClient.asCurrentUser,
         server,
-        syntheticsMonitorClient
+        syntheticsMonitorClient,
+        options
       );
 
-      const { downConfigs, staleDownConfigs, upConfigs } = await statusRule.getDownChecks(
-        ruleState.meta?.downConfigs as OverviewStatus['downConfigs']
-      );
+      const { downConfigs, staleDownConfigs, upConfigs, monitorLocationsMap } =
+        await statusRule.getDownChecks(ruleState.meta?.downConfigs as StatusConfigs);
+
+      const isCustomRule = !isEmpty(params);
 
-      Object.entries(downConfigs).forEach(([idWithLocation, { ping, configId }]) => {
-        const locationId = ping.observer.name ?? '';
-        const alertId = idWithLocation;
-        const monitorSummary = getMonitorSummary(
-          ping,
-          DOWN_LABEL,
-          locationId,
-          configId,
-          dateFormat,
-          tz
-        );
+      const { isLocationBased, downThreshold, numberOfLocations } = getConditionType(
+        params.condition
+      );
 
-        const { uuid, start } = alertsClient.report({
-          id: alertId,
-          actionGroup: MONITOR_STATUS.id,
+      const getConfigsByIds = () => {
+        const downConfigsById = new Map<string, AlertStatusMetaDataCodec[]>();
+        Object.entries(downConfigs).forEach(([_, config]) => {
+          const { configId } = config;
+          if (!downConfigsById.has(configId)) {
+            downConfigsById.set(configId, []);
+          }
+          downConfigsById.get(configId)?.push(config);
         });
-        const errorStartedAt = start ?? startedAt.toISOString();
```

Comment:
I think we should keep any previous context variables. Was this exposed on context and available for the user to use?